src/modules/Auth/components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,7 +1,131 @@
-import React from 'react'
+import React from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import styles from "./ChangePassword.module.css";
 
-export default function ChangePassword() {
-  return (
-    <div>ChangePassword</div>
-  )
+// Define the form data type
+interface ChangePasswordFormData {
+  oldPassword: string;
+  newPassword: string;
+  confirmNewPassword: string;
 }
+
+const ChangePassword: React.FC = () => {
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<ChangePasswordFormData>(); // Use the defined interface
+
+  const onSubmit = async (data: ChangePasswordFormData): Promise<void> => {
+    try {
+      const token = localStorage.getItem("token");
+
+      if (!token) {
+        toast.error("You are not authenticated. Please login first.");
+        navigate("/login");
+        return;
+      }
+
+      await axios.put(
+        "https://upskilling-egypt.com:3003/api/v1/Users/ChangePassword",
```

Comment:
We shouldn't be using hardcoded urls. We need to replace this with the axios instances and URLs objects.